core/templates/pages/splash-page/splash-page.component.ts

```diff
@@ -29,47 +30,74 @@ const splashConstants = require('constants.ts');
 angular.module('oppia').component('splashPage', {
   template: require('./splash-page.component.html'),
   controller: [
-    '$rootScope', '$timeout', 'LoaderService', 'SiteAnalyticsService',
-    'UrlInterpolationService', 'UserService', 'WindowRef',
+    '$rootScope', 'LoaderService', 'SiteAnalyticsService',
+    'UrlInterpolationService', 'UserService', 'WindowDimensionsService',
     function(
-        $rootScope, $timeout, LoaderService, SiteAnalyticsService,
-        UrlInterpolationService, UserService, WindowRef) {
+        $rootScope, LoaderService, SiteAnalyticsService,
+        UrlInterpolationService, UserService, WindowDimensionsService) {
       var ctrl = this;
       ctrl.getStaticImageUrl = function(imagePath) {
         return UrlInterpolationService.getStaticImageUrl(imagePath);
       };
-      ctrl.getStaticSubjectImageUrl = function(subjectName) {
-        return UrlInterpolationService.getStaticImageUrl(
-          '/subjects/' + subjectName + '.svg');
-      };
-
-      ctrl.onRedirectToLogin = function(destinationUrl) {
-        SiteAnalyticsService.registerStartLoginEvent(
-          'splashPageCreateExplorationButton');
-        $timeout(function() {
-          WindowRef.nativeWindow.location = destinationUrl;
-        }, 150);
-        return false;
-      };
 
       ctrl.onClickBrowseLessonsButton = function() {
         SiteAnalyticsService.registerClickBrowseLessonsButtonEvent();
-        $timeout(function() {
-          WindowRef.nativeWindow.location = (
-            `/learn/${splashConstants.DEFAULT_CLASSROOM_URL_FRAGMENT}`);
-        }, 150);
         return false;
       };
 
-      ctrl.onClickCreateExplorationButton = function() {
-        SiteAnalyticsService.registerClickCreateExplorationButtonEvent();
-        $timeout(function() {
-          WindowRef.nativeWindow.location = '/creator-dashboard?mode=create';
-        }, 150);
-        return false;
+      ctrl.getMainBackgroundUrl = function() {
+        if (ctrl.isWindowNarrow()) {
+          return ctrl.getStaticImageUrl('/splash/splashMainMobile.png');
+        } else {
+          return ctrl.getStaticImageUrl('/splash/splashMainDesktop.png');
+        }
+      };
+
+      ctrl.isWindowNarrow = function() {
+        return WindowDimensionsService.isWindowNarrow();
+      };
+
+      // The 2 functions below are to cycle between values: 0, 1, 2 for
+      // testimonialId.
+      ctrl.incrementTestimonialId = function() {
+        // This makes sure that incrementing from 2, returns 0 instead of 3,
+        // since we want the testimonials to cycle through.
+        ctrl.testimonialId = (ctrl.testimonialId + 1) % 3;
+      };
+
+      ctrl.decrementTestimonialId = function() {
+        // This makes sure that decrementing from 0, returns 2 instead of -1,
+        // since we want the testimonials to cycle through.
+        ctrl.testimonialId = (((ctrl.testimonialId - 1) % 3) + 3) % 3;
```

Comment:
```suggestion
        // This makes sure that decrementing from 0, returns 2 instead of -1,
        // since we want the testimonials to cycle through.
        ctrl.testimonialId = (ctrl.testimonialId - 1) % 3;
```
This should work too right?

Reply:
No, I had tried this, but (0 - 1) % 3 returns -1 in the JS console.